pkg/lib/kitfile/layer.go

```diff
@@ -136,32 +119,48 @@ func compressLayer(path string, mediaType constants.MediaType, ignore filesystem
 		Digest:    digester.Digest(),
 		Size:      tempFileInfo.Size(),
 	}
-	return tempFileName, desc, nil
+	layerInfo = &artifact.LayerInfo{
+		Digest: digester.Digest().String(),
+		DiffId: diffIdDigester.Digest().String(),
+	}
+	return tempFileName, desc, layerInfo, nil
 }
 
-// writeDirToTar walks the filesystem at basePath, compressing contents via the *tar.Writer.
-// Any non-regular files and directories (e.g. symlinks) are skipped.
-func writeDirToTar(basePath string, ignore filesystem.IgnorePaths, ptw *output.ProgressTar, plog *output.ProgressLogger) error {
-	// We'll want paths in the tarball to be relative to the *parent* of basePath since we want
-	// to compress the directory pointed at by basePath
-	trimPath := filepath.Dir(basePath)
-	if trimPath == "." {
-		// Avoid accidentally trimming leading `.` from filenames
-		trimPath = ""
+func writeLayerToTar(basePath string, ignore filesystem.IgnorePaths, tarWriter *output.ProgressTar, plog *output.ProgressLogger) error {
+	// Utility function to decide if two paths are in the same directory tree (i.e. one is a parent of the other)
+	sameDirTree := func(a, b string) bool {
+		aToB, errA := filepath.Rel(a, b)
+		bToA, errB := filepath.Rel(b, a)
+		if errA != nil || errB != nil {
+			plog.Logf(output.LogLevelWarn, "Cannot compare directories %s and %s, skipping path", a, b)
+			return false
+		}
+		if strings.Contains(aToB, "..") && strings.Contains(bToA, "..") {
+			return false
+		}
+		return true
 	}
```

Comment:
It might be, I opted for Contains so as to not worry about `./..`, `C:path`, etc.

I'd have to test how relative paths are handled on Windows to be sure.